cmake/ThrustBuildCompilerTargets.cmake

```diff
@@ -20,27 +31,45 @@ function(thrust_build_compiler_targets)
   if (THRUST_TBB_FOUND)
     # There's a ton of these in the TBB backend, even though the code is correct.
     # TODO: silence these warnings in code instead
-    append_option_if_available("-Wno-unused-parameter" cxx_compile_options)
+#    append_option_if_available("-Wno-unused-parameter" cxx_compile_options)
   endif()
 
   if ("MSVC" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
-    # TODO Enable /Wall instead of W3
-    append_option_if_available("/W3" cxx_compile_options)
+    append_option_if_available("/W4" cxx_compile_options)
```

Comment:
Are we sure `/W4` isn't too noisy? We tried it in the past and had to roll back.

Reply:
I had to suppress a couple of nuisance warnings, but an early draft of this patch was building cleanly with `/W4` and no warnings. So it should be fine.